use-shopping-cart/core/stripe-middleware.js

```diff
@@ -18,11 +18,36 @@ export const handleStripe = (store) => (next) => async (action) => {
       })
     }
 
-    if (store.getState().mode === 'client-only') {
+    if (cart.cartMode === 'client-only') {
       const checkoutData = getCheckoutData(cart)
       stripe.redirectToCheckout(checkoutData)
     }
   }
```

Comment:
Shouldn't `redirectToCheckout` be using two if-statements for the `cartMode` rather than checking for the `sessionId` and ignoring the `cartMode`. Also we should probably throw an error if no matching `cartMode` is found.

```
if (cart.cartMode === 'checkout-session') {
  // stripe.redirectToCheckout returns a promise which is useful if something goes wrong
  return stripe.redirectToCheckout({ /* session id stuff */ })
}
if (cart.cartMode === 'client-only') {
  // client stuff
  return stripe.redirectToCheckout(/* client stuff */)
}
throw new Error(`Invalid value for "cartMode" was found: ${cart.cartMode}`)